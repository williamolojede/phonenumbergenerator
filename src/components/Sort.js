import React from 'react'

export default function Sort({ handleSort }) {
  return (
    <div>
      <p className="uppercase mb-6 text-center text-grey-dark">Sort</p>
      <div className="flex justify-center">
        <button 
          className="border-2 border-grey-darker p-2 text-xl font-bold uppercase text-grey-darker mr-2"
          onClick={handleSort('asc')}
        >ASC</button>
        <button 
          className="border-2 border-grey-darker p-2 text-xl font-bold uppercase text-grey-darker ml-2"
          onClick={handleSort('dsc')}
        >DSC</button>
      </div>
    </div>
  )
}
