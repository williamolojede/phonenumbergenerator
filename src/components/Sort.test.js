import React from 'react';
import { mount } from 'enzyme';

import Sort from './Sort'



it('renders without crashing', () => {
  const handleSortInnerFn = jest.fn()
  const expectedCallArgs = [['asc'], ['dsc']]
  const handleSort = jest.fn((args) => handleSortInnerFn);

  const wrapper = mount(<Sort handleSort={handleSort} />);

  expect(wrapper).toMatchSnapshot();
  expect(handleSort).toHaveBeenCalledTimes(expectedCallArgs.length);
  expect(handleSort.mock.calls).toMatchObject(expectedCallArgs);

  const buttons = wrapper.find('button');
  
  buttons.at(0).simulate('click');
  buttons.at(1).simulate('click');

  expect(handleSortInnerFn ).toHaveBeenCalledTimes(2);
});
