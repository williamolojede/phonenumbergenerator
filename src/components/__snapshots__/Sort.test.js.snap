// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Sort
  handleSort={
    [MockFunction] {
      "calls": Array [
        Array [
          "asc",
        ],
        Array [
          "dsc",
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": [MockFunction],
        },
        Object {
          "isThrow": false,
          "value": [MockFunction],
        },
      ],
    }
  }
>
  <div>
    <p
      className="uppercase mb-6 text-center text-grey-dark"
    >
      Sort
    </p>
    <div
      className="flex justify-center"
    >
      <button
        className="border-2 border-grey-darker p-2 text-xl font-bold uppercase text-grey-darker mr-2"
        onClick={[MockFunction]}
      >
        ASC
      </button>
      <button
        className="border-2 border-grey-darker p-2 text-xl font-bold uppercase text-grey-darker ml-2"
        onClick={[MockFunction]}
      >
        DSC
      </button>
    </div>
  </div>
</Sort>
`;
